bin/src/main.rs

```diff
@@ -51,11 +52,52 @@ pub static PKG_NAME: &str = env!("CARGO_PKG_NAME");
 #[no_mangle]
 pub static PKG_VERSION: &str = env!("CARGO_PKG_VERSION");
 
-#[tokio::main(flavor = "multi_thread")]
-async fn main() {
+fn main() {
     env_logger::Builder::from_env(Env::default().default_filter_or("info")).init();
     info!("running version: {}", env!("CARGO_PKG_VERSION"));
 
+    // must be done at the very beginning and before other threads started
+    #[cfg(target_os = "linux")]
+    {
+        // apply capabilities only when:
+        // - k8s
+        // - docker
+        // - exe filename ends with "-no-cap" (symlinked)
+        if (std::env::var_os("KUBERNETES_SERVICE_HOST").is_some()
+            || std::path::Path::new("/.dockerenv").exists())
+            && std::env::var_os(env_vars::LOGDNA_NO_CAP).is_none()
+            && !std::env::current_exe()
+                .unwrap_or_default()
+                .ends_with("-no-cap")
+        {
+            match set_capabilities() {
+                Ok(r) if r => debug!("Using Capabilities to bypass filesystem permissions"),
+                _ => warn!("Failed to adopt capabilities to bypass DAC. The agent will only be able to access files accessible to it's user/group"),
+            }
+        }
+        let status =
+            std::fs::read_to_string("/proc/self/status").expect("Failed to read /proc/self/status");
+        let re = regex::Regex::new(r"(?m)^((Cap|Cpu|Seccomp|Groups|Uid|Gid).+?)$").unwrap();
+        for cap in re.captures_iter(status.as_str()) {
+            info!("{}", &cap[0]);
+        }
+    }
+
+    let _contents =
```

Comment:
Are these couple of lines still necessary?

Reply:
leftover